packages/babel-parser/src/parser/base.js

```diff
@@ -31,12 +31,32 @@ export default class BaseParser {
   declare input: string;
   declare length: number;
 
-  hasPlugin(name: string): boolean {
-    return this.plugins.has(name);
+  // This method accepts either a string (plugin name) or an array pair
+  // (plugin name and options object). If an options object is given,
+  // then each value is non-recursively checked for identity with that
+  // plugin’s actual option value.
+  hasPlugin(pluginConfig: PluginConfig): boolean {
+    if (typeof pluginConfig === "string") {
+      return this.plugins.has(pluginConfig);
+    } else {
+      const [pluginName, pluginOptions] = pluginConfig;
+      if (!this.hasPlugin(pluginName)) {
+        return false;
+      }
+      const actualOptions = this.plugins.get(pluginName);
+      for (const [key, value] of Object.entries(pluginOptions)) {
+        if (value !== actualOptions?.[key]) {
+          return false;
+        }
+      }
+      return true;
+    }
   }
 
   getPluginOption(plugin: string, name: string) {
     // $FlowIssue
     if (this.hasPlugin(plugin)) return this.plugins.get(plugin)[name];
```

Comment:
This is kinda off-topic, but every time I see this line I want to rewrite it with optional chaining, but it never seemed worth a PR. Perhaps you could include this change?
```js
return this.plugins.get(plugin)?.[name];
```